### PR TITLE
*: move executor/set.go special handling to setter functions

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Problem Summary:

`executor/set.go` has some special handling for setting certain variables. This has been moved to setter functions to unify how variables are set.

### What is changed and how it works?

What's Changed:

Functionality neutral change.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test (covered by existing tests)

Side effects

- None

### Release note 

- No release note

### CODE REVIEW

File: domain/sysvar_cache.go [hunk 1]
```diff
@@ -145,6 +145,10 @@ func (svc *SysVarCache) RebuildSysVarCache(ctx sessionctx.Context) error {
 }
 
 // checkEnableServerGlobalVar processes variables that acts in server and global level.
+// This is required because the SetGlobal function on the sysvar struct only executes on
+// the initiating tidb-server. There is no current method to say "run this function on all
+// tidb servers when the value of this variable changes". If you do not require changes to
+// be applied on all servers, use a getter/setter instead! You don't need to add to this list.
```
Comment (morgo) [inline domain/sysvar_cache.go:151]: This clarifies the use of this function, hoping users will not add additional cases unnecessarily. I am not sure if in future it makes sense to add functionality to specifically address this so a `SetGlobal` can execute code on all servers.

File: util/stmtsummary/variables.go [hunk 2]
```diff
@@ -104,7 +104,7 @@ func getBoolFinalVariable(varType int, sessionValue, globalValue string) int64 {
 
 // normalizeEnableValue converts 'ON' or '1' to 1 and 'OFF' or '0' to 0.
 func normalizeEnableValue(value string) int64 {
-	if svariable.TiDBOptOn(value) {
+	if strings.EqualFold(value, "ON") || value == "1" {
```
Comment (morgo) [inline util/stmtsummary/variables.go:107]: This change was required to prevent cyclic imports.

**Review by djshow832 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**